code/modules/genetics/moecubes_printer.dm

```diff
@@ -0,0 +1,182 @@
+/obj/machinery/moeballs_printer
+	name = "Gene regurgitator"
+	desc = "A stationary computer."
+	icon = 'icons/obj/eris_genetics.dmi'
+	icon_state = "printer_base"
+	density = TRUE
+	anchored = TRUE
+//	circuit = /obj/item/electronics/circuitboard/moeballs_printer
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 60
+	active_power_usage = 10000
+	req_access = list(access_research_equipment)
+	var/color_key = "yellow"
+	var/obj/item/computer_hardware/hard_drive/portable/usb
+	var/list/files[0]
+	var/index = 0
+	var/gene_cache = list(
+		"name" = "\[NOT SELECTED\]",
+		"desc" = "\[NOT FOUND\]",
+		"type" = "b_type",
+		"content" = "O-")
+
+
+/obj/machinery/moeballs_printer/initalize_statverbs()
+	if(req_access.len)
+		add_statverb(/datum/statverb/hack_console)
+
+
+/obj/machinery/moeballs_printer/Initialize(mapload, d=0)
+	. = ..()
+	color_key = default_dna_machinery_style
+	update_icon()
+
+
+/obj/machinery/moeballs_printer/New()
+	..()
+	color_key = default_dna_machinery_style
+	update_icon()
+
+
+/obj/machinery/moeballs_printer/update_icon()
+	..()
+	overlays.Cut()
+	if(!(stat & (NOPOWER|BROKEN)) && use_power)
+		var/state = "printer_idle_[color_key]"
+		overlays += state
+
+
+/obj/machinery/moeballs_printer/proc/do_flick(is_error)
+	if(is_error)
+		flick("printer_error", src)
+	else
+		flick("printer_on_[color_key]", src)
+
+
+/obj/machinery/moeballs_printer/AltClick(mob/user)
+	try_eject_usb(user)
+
+
+/obj/machinery/moeballs_printer/verb/eject_usb()
+	set name = "Eject Portable Storage"
+	set category = "Object"
+	set src in view(1)
+	try_eject_usb(usr)
+
+
+/obj/machinery/moeballs_printer/proc/try_eject_usb(mob/user)
+	if(usb && eject_item(usb, user))
```

Comment:
does this check anywhere is the user is a ghost or is not alive (stat)

Reply:
Added checks for `Adjacent()` and `user.stat`. Ghosts fail `IsAdvancedToolUser()`.